scripts/get_entities.py

```diff
@@ -0,0 +1,135 @@
+#! /usr/bin/python
+"""
+get_entities.py
+
+Usage: get_entities.py [OPTION] ... [ATTRIBUTE] ...
+
+Query the Home Assistant API for available entities then print them and any
+desired attributes to the screen.
+
+Options:
+    -h, --help                      display this text
+        --password=PASS             use the supplied password
+        --ask-password              prompt for password
+    -a, --address=ADDR              use the supplied server address
+    -p, --port=PORT                 use the supplied server port
+"""
+import sys
+import getpass
+try:
+    from urllib2 import urlopen
+    PYTHON = 2
+except ImportError:
+    from urllib.request import urlopen
+    PYTHON = 3
+import json
+
+
+def main(password, askpass, attrs, address, port):
+    """ fetch Home Assistant api json page and post process """
+    # ask for password
+    if askpass:
+        password = getpass.getpass('Home Assistant API Password: ')
+
+    # fetch API result
+    url = mk_url(address, port, password)
```

Comment:
Because that would be taking the easy route and, apparently, I like pain.

Seriously though, my mental state when I made this was to make a standalone script that did not use the code inside Home Assistant. This way it can be run from anywhere and not have to worry about messy imports.